### PR TITLE
Fix logging into dockerhub in Happy path tests

### PR DESCRIPTION
**Pull request by dmytro-ndp**
### What does this PR do?
It fixes logging into dockerhub so that it affects deployment of Eclipse Che to minikube, and also setting up Che image pull policy = ifNotPresent.

### What issues does this PR fix or reference?
#18380

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [x] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [x] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [x] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

**Comment by dmytro-ndp** [comment]
[crw-ci-test]

File: tests/.infra/crw-ci/pr-check/k8s/Jenkinsfile [hunk 1]
```diff
@@ -83,6 +83,21 @@ pipeline {
...
+                    withCredentials([string(credentialsId: 'ed71c034-60bc-4fb1-bfdf-9570209076b5', variable: 'maxura_docker_password')]) {
+                        sh """
+                            docker login -u maxura -p ${maxura_docker_password}
+
+                            TOKEN=\$(curl --user 'maxura:${maxura_docker_password}' "https://auth.docker.io/token?service=registry.docker.io&scope=repository:ratelimitpreview/test:pull" | jq -r .token)
+                            curl --head -H "Authorization: Bearer \$TOKEN" https://registry-1.docker.io/v2/ratelimitpreview/test/manifests/latest
+                        """
+                    }
+
+                    echo "Print anonymouse dockerhub limits"
```
Comment (benoitf) [inline tests/.infra/crw-ci/pr-check/k8s/Jenkinsfile]:
```suggestion
                    echo "Print anonymous dockerhub limits"
```

File: tests/.infra/crw-ci/pr-check/k8s/Jenkinsfile [hunk 2]
```diff
@@ -348,6 +391,19 @@ EOL"""
...
     post {
         cleanup {
+            withCredentials([string(credentialsId: 'ed71c034-60bc-4fb1-bfdf-9570209076b5', variable: 'maxura_docker_password')]) {
+                sh """
+                    TOKEN=\$(curl --user 'maxura:${maxura_docker_password}' "https://auth.docker.io/token?service=registry.docker.io&scope=repository:ratelimitpreview/test:pull" | jq -r .token)
+                    curl --head -H "Authorization: Bearer \$TOKEN" https://registry-1.docker.io/v2/ratelimitpreview/test/manifests/latest
+                """
+            }
+
+            echo "Print anonymouse dockerhub limits"
```
Comment (benoitf) [inline tests/.infra/crw-ci/pr-check/k8s/Jenkinsfile]:
```suggestion
            echo "Print anonymous dockerhub limits"
```

**Review by Ohrimenko1988 — APPROVED** [review]

**Comment by dmytro-ndp** [comment]
[crw-ci-test]

**Comment by dmytro-ndp** [comment]
Pre-pulling of dockerhub images being logged into docker does work:
> 7m13s       Normal    Pulled                  pod/postgres-5fff9b64c7-cn2j5                              Container image "**centos/postgresql-96-centos7:9.6**" **already present on machine**
>...
> 3m18s       Normal    Pulled                  pod/workspacels0mj82wwqbpton6.workspace-c6db97f5b-lmcr8    Container image "**mariolet/petclinic:d2831f9b**" **already present on machine**
>...
> 3m18s       Normal    Pulled                  pod/workspacels0mj82wwqbpton6.workspace-c6db97f5b-lmcr8    Container image "**centos/mysql-57-centos7**" **already present on machine**

**Pull request merged**